packages/ckeditor5-code-block/src/codeblockediting.ts

```diff
@@ -202,6 +203,31 @@ export default class CodeBlockEditing extends Plugin {
 			data.content = rawSnippetTextToViewDocumentFragment( writer, text );
 		} );
 
+		if ( editor.plugins.has( 'ClipboardPipeline' ) ) {
+			// Elements may have a plain textual representation (hence be present in the 'text/plain' data transfer),
+			// but not be allowed in the code block.
+			// Filter them out before inserting the content to the model.
+			editor.plugins.get( ClipboardPipeline ).on<ClipboardContentInsertionEvent>( 'contentInsertion', ( evt, data ) => {
+				const model = editor.model;
+				const selection = model.document.selection;
+
+				if ( !selection.anchor!.parent.is( 'element', 'codeBlock' ) ) {
+					return;
+				}
+
+				model.change( writer => {
+					const contentRange = writer.createRangeIn( data.content );
+
+					for ( const item of [ ...contentRange.getItems() ] ) {
```

Comment:
I think `for ( const item of contentRange.getItems() )` should be enough, no need to convert to array.

Reply:
There's a removal in this loop. When it triggers, the iterator stops. If you have two merge fields in the pasted content, the second one will not be deleted.